src/main/kotlin/com/noumenadigital/npl/cli/commands/CommandsRegistry.kt

```diff
@@ -0,0 +1,65 @@
+package com.noumenadigital.npl.cli.commands
+
+import com.noumenadigital.npl.cli.commands.registry.CommandExecutor
+import com.noumenadigital.npl.cli.commands.registry.HelpCommand
+import com.noumenadigital.npl.cli.commands.registry.VersionCommand
+import com.noumenadigital.npl.cli.exception.CommandNotFoundException
+import org.apache.commons.text.similarity.LevenshteinDistance
+
+enum class Command(
+    val commandName: String,
+    val description: String,
+    val createExecutor: (List<String>) -> CommandExecutor,
+) {
+    VERSION(
+        "version",
+        VersionCommand.COMMAND_DESCRIPTION,
+        { VersionCommand },
+    ),
+    HELP(
+        "help",
+        HelpCommand.COMMAND_DESCRIPTION,
+        { HelpCommand },
+    ),
+    ;
+
+    companion object {
+        fun fromString(name: String): Command? = entries.find { it.commandName.equals(name, ignoreCase = true) }
+    }
+}
+
+object CommandsRegistry {
+    fun commandFromString(
+        command: String,
+        params: List<String> = emptyList(),
+    ): CommandExecutor {
+        val normalizedCommand = command.lowercase()
+        return Command.fromString(normalizedCommand)?.createExecutor?.invoke(params)
+            ?: throw CommandNotFoundException(buildErrorMessage(normalizedCommand))
```

Comment:
For these, again, I think it would probably be better to pass in a "real" message, and then format it for printing at the catch-site.

Reply:
And maybe `CommandNotFoundException` could have a `provided` parameter taking just the command rather than just a message?